File: ios/TodayInSpaceHistoryTests/Helpers/DateExtensionTests.swift
//
//  DateExtensionTests.swift
//  TodayInSpaceHistoryTests
//

import XCTest
@testable import TodayInSpaceHistory

final class DateExtensionTests: XCTestCase {
    func testMonthIsEnglishRegardlessOfLocale() {
        var calendar = Calendar(identifier: .gregorian)
        calendar.timeZone = TimeZone(secondsFromGMT: 0)!
        let components = DateComponents(calendar: calendar, year: 2026, month: 8, day: 9)
        let date = calendar.date(from: components)!

        // Implementation must use en_US_POSIX, not the device locale.
        XCTAssertEqual(date.month, "August")
    }

    func testTodayDayMonthComponentsHasDayAndEnglishMonth() {
        let parts = Date.todayDayMonthComponents
        XCTAssertEqual(parts.count, 2)
        XCTAssertFalse(parts[0].isEmpty)

        let englishMonths: Set<String> = [
            "January", "February", "March", "April", "May", "June",
            "July", "August", "September", "October", "November", "December"
        ]
        XCTAssertTrue(
            englishMonths.contains(parts[1]),
            "Expected en_US_POSIX month name, got \(parts[1])"
        )
    }
}

File: android/app/src/test/java/miko/todayinspacehistory/helpers/DateExtensionTests.kt
package miko.todayinspacehistory.helpers

import miko.todayinspacehistory.util.monthName
import miko.todayinspacehistory.util.todayDayMonthComponents
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import java.util.Calendar
import java.util.GregorianCalendar

class DateExtensionTests {
    @Test
    fun monthIsEnglishRegardlessOfLocale() {
        val calendar = GregorianCalendar()
        calendar.set(2026, Calendar.AUGUST, 9, 12, 0, 0)
        val date = calendar.time

        assertEquals("August", date.monthName())
    }

    @Test
    fun todayDayMonthComponentsHasDayAndEnglishMonth() {
        val parts = todayDayMonthComponents()
        assertEquals(2, parts.size)
        assertFalse(parts[0].isEmpty())

        val englishMonths = setOf(
            "January", "February", "March", "April", "May", "June",
            "July", "August", "September", "October", "November", "December",
        )
        assertTrue(
            "Expected English month name, got ${parts[1]}",
            englishMonths.contains(parts[1]),
        )
    }
}
